node/modules/actorevent.go

```diff
@@ -0,0 +1,138 @@
+package modules
+
+import (
+	"context"
+	"time"
+
+	"github.com/multiformats/go-varint"
+	"go.uber.org/fx"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	builtintypes "github.com/filecoin-project/go-state-types/builtin"
+
+	"github.com/filecoin-project/lotus/chain/events"
+	"github.com/filecoin-project/lotus/chain/events/filter"
+	"github.com/filecoin-project/lotus/chain/messagepool"
+	"github.com/filecoin-project/lotus/chain/stmgr"
+	"github.com/filecoin-project/lotus/chain/store"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/node/config"
+	"github.com/filecoin-project/lotus/node/impl/full"
+	"github.com/filecoin-project/lotus/node/modules/helpers"
+)
+
+type EventAPI struct {
+	fx.In
+
+	full.ChainAPI
+	full.StateAPI
+}
+
+var _ events.EventAPI = &EventAPI{}
+
+func EthEventAPI(cfg config.ActorEventConfig) func(helpers.MetricsCtx, fx.Lifecycle, *store.ChainStore, *stmgr.StateManager, EventAPI, *messagepool.MessagePool, full.StateAPI, full.ChainAPI) (*full.EthEvent, error) {
+	return func(mctx helpers.MetricsCtx, lc fx.Lifecycle, cs *store.ChainStore, sm *stmgr.StateManager, evapi EventAPI, mp *messagepool.MessagePool, stateapi full.StateAPI, chainapi full.ChainAPI) (*full.EthEvent, error) {
+		ee := &full.EthEvent{
+			Chain:                cs,
+			MaxFilterHeightRange: abi.ChainEpoch(cfg.MaxFilterHeightRange),
+		}
+
+		if !cfg.EnableRealTimeFilterAPI {
+			// all event functionality is disabled
+			// the historic filter API relies on the real time one
+			return ee, nil
+		}
+
+		ee.SubManager = &full.EthSubscriptionManager{
+			Chain:    cs,
+			StateAPI: stateapi,
+			ChainAPI: chainapi,
+		}
+		ee.FilterStore = filter.NewMemFilterStore(cfg.MaxFilters)
+
+		// Start garbage collection for filters
+		lc.Append(fx.Hook{
+			OnStart: func(ctx context.Context) error {
+				go ee.GC(ctx, time.Duration(cfg.FilterTTL))
```

Comment:
The context in `OnStart` is only alive during node startup (for ~5 seconds), we probably want the context we get from `ctx := helpers.LifecycleCtx(mctx, lc)` below